flycheck.el

```diff
@@ -6924,7 +6924,16 @@ See URL `http://www.erlang.org/'."
 
 (defun flycheck-rebar3-project-root (&optional _checker)
   "Return directory where =rebar.config= is located."
-  (locate-dominating-file buffer-file-name "rebar.config"))
+  (let* ((rebarconfig (locate-dominating-file buffer-file-name "rebar.config"))
+         (project-dir (if rebarconfig
+                          (file-name-as-directory
+                           (substring rebarconfig
+                                      0
+                                      (string-match
+                                       "\\(/_build/[^/]+/lib\\|/_checkouts\\)"
+                                       rebarconfig)))
```

Comment:
I like it. The issue with step 2 is that I need to look for `_build` or `_checkouts`; however,  there's an even better solution. We can just keep looking for `rebar.config`. Every `rebar3` project needs one, so a recursive call to `locate-dominating-file` would be enough. I've rust run into one snag: A recursive call to `locate-dominating-file` finds the one in `.`, so I need the recursive call to run on the parent directory, and I just can't figure out the function to give me the parent-dir, do you know how to do that? Thanks for the `elisp` lesson :D